using-rest-api.md updated (#207)
Macrometacorp/docs: docs/collections/graphs/using-rest-api.md

  <TabItem value="py" label="Python">

```py
""" This file is a demo on using edge collections and graphs """
import json
import requests

# Constants

FEDERATION = "api-gdn.paas.macrometa.io"
FED_URL = f"https://{FEDERATION}"
GEO_FABRIC = "_system"

> API_KEY = "my IP key" #Change to my API key

Reviewer: ```suggestion
API_KEY = "my IP key" # Change to your API key
```

#EMAIL = "my email"
#PASSWORD = "my password"

> #AUTH_TOKEN = "bearer " #do not change

Reviewer: ```suggestion
#AUTH_TOKEN = "bearer " # Do not change
```

> #AUTH_TOKEN = "bearer " # Do not change

Reviewer 2: we should not keep this email,pass, authtoken commented since we are not showing the user any alternative way to use them here. if we are using APikey  then we should remove these other methods. Also API_KEY should have sample like API_KEY = "apikey yourapikey" since "apikey" keyword is needed and user doesn't know that. Really sorry for being so late.

Author: Hi. I left these available in case the user chooses that method instead of the API key way. The user should also uncomment the multi-line comment beginning at line 84-85 "url = f"{FED_URL}/_open/auth"".

The "apikey" keyword is not needed as that works on my tests.

Thanks for your comments.

COLLECTION_NAME_1 = "teachers"
COLLECTION_NAME_2 = "lectures"
EDGE_COLL_NAME = "teach"
GRAPH_NAME = "lectureteacher"

# Create a HTTPS Session Part 1

> # This part is needed when getting access through email/pass instead of API key or JWT token

Reviewer: ```suggestion
# The following is required when accessing Macrometa with your email and password instead of an API key or JWT token
```

"""""

> url = f"{FED_URL}/_open/auth"

Reviewer 2: Oops I might have missed this while reviewing the audit code... but we don't need to call /_open/auth endpoint here since we are using Api-key now, auth endpoint is used to get JWT token by logging in from email and password. Really sorry for the late comment.

Author: Hi Koshy.

Yes, I left that in so a user can use both possibilities, either an API key or email/pass. The latter is behind a multi-line comment, beginning with the URL variable... mmm constant. I'll check for PEP8 compliance on this part of the code.

Reviewer 2: <img width="320" alt="image" src="https://user-images.githubusercontent.com/107413064/185224215-0d2cf82f-2828-472b-9f11-4e7f7fad8a90.png">
I guess you added 2 extra double quotes so I guess its not a comment anymore?

Author: OK, that part of the code has been updated.
Got a couple of unneeded double quotes.

Thanks for your review Koshy.

payload = {
    'email':EMAIL,
    'password':PASSWORD
}
headers = {
    'content-type': 'application/json'
}

response = requests.post(url, data = json.dumps(payload), headers = headers)
print("Response code:", response.status_code)
if response.status_code == 200:
    resp_body = json.loads(response.text)
    AUTH_TOKEN += resp_body["jwt"]
    TENANT = resp_body["tenant"]
else:
    raise Exception(f"Error while getting auth token. Code:{response.status_code},
        Reason:{response.reason}")
print("Auth token:", AUTH_TOKEN)

"""
# Create a HTTPS Session Part 2
session = requests.session()
session.headers.update({"content-type": 'application/json'})
session.headers.update({"authorization": "apikey " + API_KEY})

> # Create Doc Collections and Insert Data to Document Collections

Reviewer: ```suggestion
# Create Document Collections and Insert Data
```

resp = session.post(url,data=json.dumps(payload))
result = json.loads(resp.text)
print("\nEdge Collection Created: ",result)

payload = [
    {
    '_key': 'Jean-CSC101',
    '_from': 'teachers/Jean',
    '_to': 'lectures/CSC101',
    'online': False
    },
    {
    '_key': 'Jean-CSC102',
    '_from': 'teachers/Jean',
    '_to': 'lectures/CSC102',
    'online': True
    },
    {
    '_key': 'Jean-CSC103',
    '_from': 'teachers/Jean',
    '_to': 'lectures/CSC103',
    'online': False
    },
    {
    '_key': 'Bruce-CSC101',
    '_from': 'teachers/Bruce',
    '_to': 'lectures/CSC101',
    'online': True
    }
]

url = f"{FED_URL}/_api/document/{EDGE_COLL_NAME}"
resp = session.post(url,data=json.dumps(payload))
result = json.loads(resp.text)
print("\nDocuments Inserted: ",result)
# Create a Graph
payload = {
    "edgeDefinitions": [
    {
    "collection": EDGE_COLL_NAME,
    "from": [ "teachers" ],
    "to": [ "lectures" ]
    }
    ],
    "name": GRAPH_NAME,
    "options": {}
}

url = f"{FED_URL}/_api/graph"
resp = session.post(url,data=json.dumps(payload))
result = json.loads(resp.text)
print("\nGraph Created: ",result)

# Graph Traversal

> # Note :- To use Outbound Traversal use direction: out and direction: in for Inbound Traversal

Reviewer: ```suggestion
# Set `direction` to `out` or `in` for outbound or inbound traversal.
```

params = {
    "vertex": "teachers/Jean",
    "direction": "out"
}

url = f"{FED_URL}/_api/edges/{EDGE_COLL_NAME}"

resp = session.get(url,params=params)
result = json.loads(resp.text)
print("\nGraph Traversal: ",result)

# Delete Graph and Collections

> # Note:- If you want to delete just the graph and keep collections then

Reviewer: ```suggestion
# Set `dropCollection` to `False` if you want to delete the graph but keep the collections.
```
